src/main/kotlin/uk/gov/justice/hmpps/offenderevents/service/OffenderEventStore.kt

```diff
@@ -14,8 +14,13 @@ class OffenderEventStore(@Value("\${model.cacheSize}") private val cacheSize: In
     fun fromJson(json: String): MutableMap<String, String> {
       return try {
         Gson().fromJson(json, object : TypeToken<MutableMap<String, String>>() {}.type)
-      } catch(e: Exception) {
-        mutableMapOf("BadMessage" to json, "CausedException" to e.toString())
+      } catch (e1: Exception) {
+        try {
+          val anyMap: Map<String, Any> = Gson().fromJson(json, object : TypeToken<Map<String, Any>>() {}.type)
+          anyMap.entries.associate { it.key to it.value.toString() }.toMutableMap()
```

Comment:
If the values in the JSON can be deserialized as type Any, we convert them to string and send them back.